src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,180 @@
-package core.basesyntax;
-
-import java.util.List;
-
-public class MyLinkedList<T> implements MyLinkedListInterface<T> {
-    @Override
-    public void add(T value) {
-    }
-
-    @Override
-    public void add(T value, int index) {
-    }
-
-    @Override
-    public void addAll(List<T> list) {
-    }
-
-    @Override
-    public T get(int index) {
-        return null;
-    }
-
-    @Override
-    public T set(T value, int index) {
-        return null;
-    }
-
-    @Override
-    public T remove(int index) {
-        return null;
-    }
-
-    @Override
-    public boolean remove(T object) {
-        return false;
-    }
-
-    @Override
-    public int size() {
-        return 0;
-    }
-
-    @Override
-    public boolean isEmpty() {
-        return false;
-    }
-}
+package core.basesyntax;
+
+import java.util.List;
+import java.util.Objects;
+
+public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+
+    private Node head;
+    private Node tail;
+    private int size;
+
+    public class Node<T> {

```

Comment:
we shouldn't expose inner class in this case

Reply:
not fixed, wrong access modifier. also move it to the very end of the file